src/internal/types.rs

```diff
@@ -276,3 +276,87 @@ impl TransactionId {
         TransactionId(id)
     }
 }
+
+#[derive(Copy, Clone, Debug)]
+pub enum ReadDirection {
+    Forward,
+    Backward,
+}
+
+#[derive(Debug)]
+pub enum LocatedEvents<A> {
+    EndOfStream,
+    Events { events: Vec<ResolvedEvent>, next: Option<A> },
+}
+
+pub trait Slice {
+    type Location;
+
+    fn from(&self) -> Self::Location;
+    fn direction(&self) -> ReadDirection;
+    fn events(self) -> LocatedEvents<Self::Location>;
+}
+
+#[derive(Debug)]
+pub enum ReadStreamStatus<A> {
+    Success(A),
+    NoStream(String),
+    StreamDeleled(String),
+    NotModified(String),
+    Error(String),
+    AccessDenied(String),
+}
+
+#[derive(Debug)]
+pub struct StreamSlice {
+    _from: i64,
+    _direction: ReadDirection,
+    _events: Vec<ResolvedEvent>,
+    _next_num_opt: Option<i64>,
```

Comment:
Why did you prefix fields with underscores ?

Reply:
I saw some Rust codebase where private fields were prefixed with an underscore. I don't know if I will maintain that guideline at the end.

Reply:
underscores are used to tell the compiler that you don't use variables/fields and avoid warning messages, as in Haskell.